### PR TITLE
feat(performance): Add team key transaction field

### PR DESCRIPTION
**Pull request by Zylphrex**
This adds a team_key_transaction column to query team based key transactions.

### CODE REVIEW

File: src/sentry/search/events/fields.py [hunk 1]
```diff
@@ -112,6 +113,45 @@ def key_transaction_expression(user_id, organization_id, project_ids):
     ]
 
 
+def team_key_transaction_expression(organization_id, team_ids, project_ids):
+    if organization_id is None or team_ids is None or project_ids is None:
+        raise InvalidSearchQuery("Missing necessary meta for key transaction field.")
```
Comment (wmak) [inline src/sentry/search/events/fields.py:118]: Hmm can we reword this to be something more user friendly? Actually is it possible for a user to cause this? If one of these attributes are missing its likely a bug right?

File: src/sentry/search/events/fields.py [hunk 2]
```diff
@@ -112,6 +113,45 @@ def key_transaction_expression(user_id, organization_id, project_ids):
...
+    team_key_transactions = (
+        TeamKeyTransaction.objects.filter(
+            organization_id=organization_id,
+            project_id__in=project_ids,
+            team_id__in=team_ids,
+        )
+        .order_by("transaction", "project_id")
+        .values("project_id", "transaction")
+        .distinct("transaction", "project_id")
+    )
```
Comment (wmak) [inline src/sentry/search/events/fields.py:129]: Not sure if this is planned in a future part, but should this query have a limit to avoid hitting the max query size?

File: src/sentry/search/events/fields.py [hunk 3]
```diff
@@ -112,6 +113,45 @@ def key_transaction_expression(user_id, organization_id, project_ids):
...
+
+    if not len(team_key_transactions):
```
Comment (wmak) [inline src/sentry/search/events/fields.py:131]: nit: but why not just check its 0 instead of going with a falsey check?
```suggestion
    if len(team_key_transactions) == 0:
```

File: src/sentry/search/events/fields.py [hunk 4]
```diff
@@ -112,6 +113,45 @@ def key_transaction_expression(user_id, organization_id, project_ids):
...
+        return ["toInt64", [0]]
```
Comment (wmak) [inline src/sentry/search/events/fields.py:132]: Could we comment that this is to get False?
Based on the docs this should be `toInt8` right? https://clickhouse.tech/docs/en/sql-reference/data-types/boolean/

File: src/sentry/search/events/filter.py [hunk 5]
```diff
@@ -627,11 +650,13 @@ def get_filter(query=None, params=None):
...
         for key in ("start", "end"):
             kwargs[key] = params.get(key, None)
-        # OrganizationEndpoint.get_filter() uses project_id, but eventstore.Filter uses project_ids
         if "user_id" in params:
             kwargs["user_id"] = params["user_id"]
         if "organization_id" in params:
             kwargs["organization_id"] = params["organization_id"]
+        if "team_id" in params:
+            kwargs["team_id"] = params["team_id"]
+        # OrganizationEndpoint.get_filter() uses project_id, but eventstore.Filter uses project_ids
```
Comment (wmak) [inline src/sentry/search/events/filter.py:659]: :clap: ty

**Review by wmak — APPROVED** [review]
Mostly minor comments but looks good to me, my biggest concern is with the lack of `limit` on the query, but this is still internal so not a blocker

File: src/sentry/search/events/fields.py (hunk 1 above)
Comment (Zylphrex) [reply src/sentry/search/events/fields.py:118]: Yeah, if this error ever happens it should be a bug, I'll update this to 500 instead so the error is captured.

File: src/sentry/search/events/fields.py (hunk 2 above)
Comment (Zylphrex) [reply src/sentry/search/events/fields.py:129]: Was going to leave the limit for for a future PR
